Sources/SteamPress/Provider.swift

```diff
@@ -1,57 +1,79 @@
 import Vapor
 import Fluent
-import Paginator
-import LeafMarkdown
+import MarkdownProvider
+import LeafProvider
+import AuthProvider
+import Sessions
+import Cookies
+import Foundation
 
 public struct Provider: Vapor.Provider {
 
-    public var provided: Providable = Providable()
-
     private static let configFilename: String = "steampress"
+    public static let repositoryName: String = "steampress"
     
     private let blogPath: String?
     private let postsPerPage: Int
     private let pathCreator: BlogPathCreator
     private let useBootstrap4: Bool
     private let enableAuthorsPages: Bool
     private let enableTagsPages: Bool
-
-    public func boot(_ drop: Droplet) {
-
-        setup(drop)
-        
-        let viewFactory = LeafViewFactory(viewRenderer: drop.view)
-
-        // Set up the controllers
-        let blogController = BlogController(drop: drop, pathCreator: pathCreator, viewFactory: viewFactory, postsPerPage: postsPerPage, enableAuthorsPages: enableAuthorsPages, enableTagsPages: enableTagsPages, config: drop.config)
-        let blogAdminController = BlogAdminController(drop: drop, pathCreator: pathCreator, viewFactory: viewFactory, postsPerPage: postsPerPage)
-
-        // Add the routes
-        blogController.addRoutes()
-        blogAdminController.addRoutes()
-    }
     
-    func setup(_ drop: Droplet) {
+    public func boot(_ config: Config) throws {
+        try config.addProvider(AuthProvider.Provider.self)
+        
         // Database preperations
```

Comment:
The comment should be "Database preparations" not "Database preperations". It's misspelled. ;-)